src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageMetadataReadMeControl.xaml.cs

```diff
@@ -19,11 +15,84 @@ namespace NuGet.PackageManagement.UI
     /// <summary>
     /// Interaction logic for PackageMetadataReadMeControl.xaml
     /// </summary>
-    public partial class PackageMetadataReadMeControl : UserControl
+    public partial class PackageMetadataReadMeControl : UserControl, IDisposable
     {
+#pragma warning disable CS0618 // Type or member is obsolete
+        private IMarkdownPreview _markdownPreview;
+#pragma warning restore CS0618 // Type or member is obsolete
+        private bool _disposed = false;
+
+        [Obsolete]
         public PackageMetadataReadMeControl()
         {
             InitializeComponent();
         }
+
+        public void Dispose()
+        {
+            Dispose(disposing: true);
+            GC.SuppressFinalize(this);
+        }
+
+        protected virtual void Dispose(bool disposing)
+        {
+            if (_disposed)
+            {
+                return;
+            }
+
+            if (disposing)
+            {
+                _markdownPreview?.Dispose();
+                var viewModel = (ReadMePreviewViewModel)DataContext;
+                viewModel.PropertyChanged -= ViewModel_PropertyChangedAsync;
+            }
+
+            _disposed = true;
+        }
+
+        private void PackageMetadataReadMeControl_DataContextChanged(object sender, DependencyPropertyChangedEventArgs e)
+        {
+            var viewModel = (ReadMePreviewViewModel)DataContext;
+            viewModel.PropertyChanged += ViewModel_PropertyChangedAsync;
```

Comment:
This presents a memory leak. Mentioned in another comment, when you register this new view model to listen for events, you have to unregister the old one first or it will continue to hold on to a reference to this control and it will never be able to be garbage collected.